kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/AbstractRefinableView.java

```diff
@@ -223,4 +223,41 @@ public URI getUri() {
     }
     return builder.build();
   }
+
+  @Override
+  public void signalReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
+
+    repo.signalReady(dataset.getNamespace(), dataset.getName(), getConstraints());
+  }
+
+  @Override
+  public boolean isReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
```

Comment:
Why not have the dataset manage its own signals? I don't see a need to manage the signals at a repository/provider level because there aren't many operations that make sense. It seems like you will always want to get a view and then check if it is ready:

```java
Datasets.load("view:...").isReady();
```